### ISSUE TITLE
Safari "Allow websites to check for Apple Pay" setting has to be disabled in order for Apple Pay button to appear.

### ISSUE THREAD

**Issue by csmcneill** [Automattic/woocommerce-payments#3876]
### Describe the bug

If `Allow websites to check for Apple Pay and Apple Card` is checked in **Safari > Preferences > Privacy**, Apple Pay may not display on sites using WooCommerce Payments.

### To Reproduce

1. Use Safari with the setting enabled
2. Open the sample URLs in private mode (the button expected to appear on both sites)
3. Open the sample URLs in normal mode (the button expected to appear only on one site)

### Actual behavior

Apple Pay will not display, even though it is enabled and configure properly.

### Screenshots

With this option enabled:

![https://d.pr/i/lWPplj](https://d.pr/i/lWPplj+) 
Full Size: https://d.pr/i/lWPplj

With this option disabled:

![https://d.pr/i/HBkCzr](https://d.pr/i/HBkCzr+) 
Full Size: https://d.pr/i/HBkCzr

### Expected behavior

Apple Pay should appear when this option is checked.

### Desktop (please complete the following information):

* OS: macOS
* Browser: Safari
* Version: 15.2 (17612.3.6.1.6)

### Additional context

See [#1111 on the Stripe repo](https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1111).

Slack threads: 

- p1619724791069000-slack-C7U3Y3VMY (see the thread for sample URLs)
- p1646161452898609-slack-C7U3Y3VMY

Ticket: 4822124-zd-woothemes

**Comment by aheckler** [comment]
4837851-zen

**Comment by jessepearson** [comment]
I received this reply last night from Stripe:
> You should be able to delete the domain by calling DELETE on /v1/apple_pay/domains/{domain}.

This means it may be possible to develop a mechanism to "reset" the Apple pay domain for an express account.

**Comment by dreamtooloud** [comment]
Pretty confident we have another report here : 5833045-zen

**Comment by drwpcom** [comment]
@jessepearson Can we reconsider this priority given that there is no workaround?

**Comment by zoemyers** [comment]
Got another one here: [5958049-zd](https://woothemes.zendesk.com/agent/tickets/5958049)

**Comment by zoemyers** [comment]
Looking for an update on priority for this issue, since it's been a few weeks since it was asked.

**Comment by jessepearson** [comment]
Sorry, I had thought I replied. I added this to our team's backlog so that we can take a look at it during a cooldown week. Next week just happens to be a cooldown week and also my janitorial rotation, so I will try to look into seeing if what I mentioned before is possible.

**Comment by zoemyers** [comment]
Hey @jessepearson - just checking in to see how things went this week with this issue?

**Comment by jessepearson** [comment]
Due to we are behind on a project, our cooldown week was skipped and my janitorial rotation was also swapped with someone else.

Can you confirm that this is still an issue with the above customers?

**Comment by Brianmitchtay** [comment]
Seeing this occur again in 6301929-zen. I would love to see this prioritized and resolved since it's highly counterintuitive, merchants might lose hours trying to troubleshoot the issue via other means (as this user had), and are quite unlikely to find this bug report.

**Comment by Brianmitchtay** [comment]
6325683-zen

Another user who had dug through all the troubleshooting in the documentation and was at a loss until Happiness informed them of the bug. I think we may need to add this bug to the documentation to decrease merchant frustration and support load until a fix is implemented.

**Comment by aheckler** [comment]
Added a note about this here: https://woocommerce.com/document/woocommerce-payments/payment-methods/apple-pay/#button-does-not-appear

**Comment by zoemyers** [comment]
6539281-zen Can we get an update and a higher priority on this?

**Comment by jessepearson** [comment]
@zoemyers I am not sure which team would actually handle something like this. I initially looked into it to see if it was possible, and it should be possible to create a tool in MC for HEs to help out here, but it would need to be built to be tested.

My team's next cooldown week is Aug 7-11, and that's where we can work on things outside of our current project work. I've set up a couple reminders for myself to take a look that week.

**Comment by jessepearson** [comment]
I started looking into this due to blockers on the current project. I have found that it may be possible to add a button in WooCommerce > Status > Tools that could reregister the Apple Pay domain for the merchant. I can create a POC (proof of concept) for this, then run it by leads to see if it is okay. Unfortunately, I won't be able to look at this again until the time frame I mentioned before due to AFK.

One of the pitfalls is Apple Pay only works on live accounts, not test accounts. This means I will have to create a live account to test with, and with my AFK next week and JN sites expiring in a week, I don't want to start that process now. I will create some of the needed code, though, so it will be moving forward.

Steps:
- [ ] Create a site with a live WCPay account to be able to test with.
- [ ] Confirm if the [`register_domain_with_apple()` method](https://github.com/Automattic/woocommerce-payments/blob/45f6a80c5870af32b2fbb0fb420c988252949dae/includes/class-wc-payments-apple-pay-registration.php#L262) is able to register the domain again.
  - I wasn't able to find API documentation around creating or deleting Apple Pay domains, so I am not sure if you can `CREATE` a domain again and it replaces the currently existing one.
- [ ] If the above method does not work, then a new method will need to be created to delete the domain.
- [ ] An endpoint will need to be created on the server to receive the delete request.
- [ ] A new method will need to be created to call the new delete method, and then call the register method to recreate it.
- [ ] Confirm this works on a live account, it deletes the domain, and then adds the "new" one.

There may be additional steps, but this is the gist of it.

**Comment by tpaksu** [comment]
@Automattic/gamma feel free to reassign this issue to the related team. Thanks!

**Comment by htdat** [comment]
@anu-rock (team Fusion) - could you have a look at this? It's a payment-request issue. 

I am removing `priority:low` label as we have quite a lot of affected merchants.

**Comment by elazzabi** [comment]
Wanted to replicate this so I could investigate... but

<img width="1724" alt="Privacy_and_Checkout_–_WooCommerce_Payments_Dev" src="https://github.com/Automattic/woocommerce-payments/assets/12542046/4cb4dc24-384e-46fc-aaec-5e104806098d">

The Apple Pay button appears for me. Are there any other steps I need to replicate this issue? @csmcneill / @aheckler or anyone who was able to replicate this 🙇

**Comment by aheckler** [comment]
I'm honestly not sure how to repro now. :/ I just checked my test site, and Apple Pay shows up for me in Safari regardless of whether or not I have "Allow websites to check for Apple Pay" checked. 🤷‍♂️

**Comment by jessepearson** [comment]
That's always been one of the hardest things to do regarding this issue, even before WooPayments existed, reproducing this bug.

My initial thought was creating an internal tool to allow support to delete the domain to allow it to be registered again automatically, but I didn't want to add any additional support load. From my research, it seems like we could add a button under WooCommerce > Status > Tools to do a "reconnect" of sorts. The button could call the API to delete the domain, and then it should be automatically registered again when the page refreshes (through existing functionality).

**Comment by elazzabi** [comment]
Forgot to mention that we are continuing this discussion on [Slack](https://a8c.slack.com/archives/C04DN7Q51KK/p1697443672595599) to speed up things and make a decision about the next steps. I'll mark this as blocked until we have a clear idea about next steps

**Comment by pierorocca** [comment]
Thanks @elazzabi. Enjoy the sabbatical and I'll pick this up next week.

**Comment by pierorocca** [comment]
Same here I cannot replicate with the Safari settings. If it does come up again on a site with a suspect domain registration issue, try toggling the settings and see if it has an effect. Perhaps that setting is only pertinent when the domain is not registered/trusted.

**Comment by pierorocca** [comment]
Testing all HE tickets reported:

> 4837851-zen

Apple Pay button displays.

> 5833045-zd-a8c

Apple Pay button displays.

> 5958049-zd-a8c

Apple Pay button displays.

> 6301929-zd-a8c

@Brianmitchtay the store is hosted on a subdomain /shop thus the merchant will need to manually place the domain association file in the .well-known folder on the root domain. I checked their site and Apple is not rendering causing UI issues.

> 6539281-zd-a8c

Apple Pay does not display. However the ticket indicates a 500 error which means Apple is unable to access the domain registration file. This is not related to a bug @zoemyers. Stripe shows 12 failed registration attempts in July.

> 6325683-zd-a8c

Apple Pay button is visible. They've got a white button on a white background. @Brianmitchtay might be worth advising them to switch to a Dark button profile rather than the light profile.


To summarize:

1. The "Allow websites to check for Apple Pay" is not an issue. That's a feature of Safari 17 that allows non-domain registered sites to check for Apple Pay.
2. All sites reported in this GH issue either now have Apple Pay displaying except two: 1) Store is installed on /shop subdomain. The domain association file is therefore not on the root. This is expected. 2) Store experiences a 500 error which means it's an access issue not a bug.

I'm going to go ahead and close this issue as not reproducible (settings), no longer issues (sites have button displayed, and not bugs.

**Comment by pierorocca** [comment]
@frosso even with the light profile, wouldn't the button have a border? 6325683-zd-a8c I see none on the page which is odd.

![image](https://github.com/Automattic/woocommerce-payments/assets/109928953/1844ec56-948e-4ee3-92ae-192264cfa068)

![image](https://github.com/Automattic/woocommerce-payments/assets/109928953/43836c19-6b0a-46f6-ac7e-042b03e42462)

**Comment by pierorocca** [comment]
Oh I see, the merchant has both Stripe and WooPayments gateways active. 

Two issues present related to the Stripe gateway:
1. There's no border around the light profile buttons
2. Quick win opportunities to improve the shortcode buttons as was done for WooPayments per recent discussions

cc @diegocurbelo @james-allan
